clock/clock.go

```diff
@@ -0,0 +1,54 @@
+package clock
+
+import (
+	"sync"
+	"time"
+)
+
+// Clock defines an interface for fetching time that may be used instead of the
+// time module.
+type Clock interface {
+	Now() time.Time
+}
+
+// BlockClock delegates calls to the time package.
+type BlockClock struct{}
+
+// NewBlockClock returns a BlockClock that delegates calls to the time package.
+func NewBlockClock() *BlockClock {
+	return &BlockClock{}
+}
+
+// Now returns the current local time.
+func (bc *BlockClock) Now() time.Time {
+	return time.Now()
+}
+
+// MockClock returns a mocked clock implementation that may be manually
+// set for testing things related to time.
+type MockClock struct {
```

Comment:
Should this belong in its own file, and perhaps in its own package that has to do with testing/test helpers?